### ISSUE TITLE
Sampling off session replays based off device.class for iOS/Android SDKs

### ISSUE THREAD

**Issue by sentryadam0000345** [getsentry/sentry-cocoa#4387]
### Problem Statement

Currently this is not supported based off the docs listed [here](https://docs.sentry.io/platforms/apple/guides/ios/session-replay/#sampling).

### Solution Brainstorm

Having something similiar to the transactions sampler that we have here would be ideal:

[https://docs.sentry.io/platforms/javascript/configuration/sampling/#setting-a-sampling-function](https://docs.sentry.io/platforms/javascript/configuration/sampling/#setting-a-sampling-function)

### Product Area

Replays

**Comment by jas-kas** [comment]
Hey @sentryadam0000345 👋

Thanks for your feature request. This work is not currently scoped as part of our general availability release but we'll log this in our backlog.

Is there a particular use-case where you'd find this useful?

**Comment by sentryadam0000345** [comment]
With potentially higher performance impacts with the new mobile session replays product, I might want to make sure that none of my lower class devices have mobile session replay turned on or are collected (so perhaps this is more about session replay being selectively initialized for a certain class of devices).

**Comment by bruno-garcia** [comment]
Is the goal to not record a replay of the overhead can be noticeable to the user? If so, we might have something built in that turns off replay if we detect the overhead is high. But a custom sampling strategy based on device class has some limitations. The main being the fact we don't know the class in the device. That's populated server side in Sentry.

**Comment by sentryadam0000345** [comment]
Yes that's the goal and turning off replay if overhead is high sounds like it can work.  Thanks
